crates/lib/src/relation.rs

```diff
@@ -92,16 +93,26 @@ pub struct Header {
     pub table_id: TableId,
     pub table_name: Box<str>,
     pub fields: Vec<Column>,
-    pub constraints: Vec<(ColList, Constraints)>,
+    pub constraints: BTreeMap<ColList, Constraints>,
 }
 
 impl Header {
+    /// Create a new header.
+    /// Note that equal ColLists with different Constraints will have their constraints unioned.
     pub fn new(
         table_id: TableId,
         table_name: Box<str>,
         fields: Vec<Column>,
-        constraints: Vec<(ColList, Constraints)>,
+        uncompressed_constraints: impl IntoIterator<Item = (ColList, Constraints)>,
```

Comment:
```suggestion
       proposed_constraints: impl IntoIterator<Item = (ColList, Constraints)>,
```

I think `compression` conveys a different thing.

Reply:
I reworked the structure here a little and changed this to `uncombined_constraints`. See the doc comment on `relation.rs:combine_constraints`.